### PR TITLE
83 post commentscomment idlikes

### PR DESCRIPTION
**Pull request by DLiSWE**
Create Comment Endpoint created

### CODE REVIEW

**Review by Matt-Tan15 — APPROVED** [review]

File: src/services/comment.service.ts [hunk 1]
```diff
@@ -42,6 +50,10 @@ class CommentService {
     });
     return getLikes?.[0].likes || [];
   }
+
+  public async likeComment({ id, userId }: any): Promise<void> {
+    await pgDataSource.createQueryBuilder().relation(CommentEntity, 'likes').of(id).add(userId);
```
Comment (raylu511) [inline src/services/comment.service.ts:55]: Do we want to check if the user already liked the post?

Comment (raylu511) [reply src/services/comment.service.ts:55]: Can you also explain to me what this query builder does?

File: src/controllers/comment.controller.ts [hunk 2]
```diff
@@ -69,6 +69,17 @@ class CommentController {
       next(error);
     }
   };
+
+  public likeComment = async (req: RequestWithUser, res: Response, next: NextFunction) => {
+    try {
+      const id = Number(req.params.comment_id);
+      const userId = req.user.id;
+      await this.commentService.likeComment({ id, userId });
+      res.status(200).json('Comment Liked');
```
Comment (ZohaibManzoor00) [inline src/controllers/comment.controller.ts:78]: status code should be 201. On google it says 201 if the "request has succeeded and has led to the creation of a resource".

File: src/services/comment.service.ts [hunk 3]
```diff
@@ -42,6 +50,10 @@ class CommentService {
     });
     return getLikes?.[0].likes || [];
   }
+
+  public async likeComment({ id, userId }: any): Promise<void> {
```
Comment (Matt-Tan15) [inline src/services/comment.service.ts:54]: { id, userId }: any should use a DTO

File: src/services/comment.service.ts (hunk 1 above)
Comment (DLiSWE) [reply src/services/comment.service.ts:55]: So the createQueryBuilder method has a relation method there it takes the Entity where the many to many unidirectional field is and looks for the name 'likes' or whatever the field name is. ".of(id)" refers to the comment_id where we are adding the userId.

Comment (DLiSWE) [reply src/services/comment.service.ts:55]: https://typeorm.io/relational-query-builder

**Review by ZohaibManzoor00 — DISMISSED** [review]
Looks good

File: src/services/comment.service.ts [hunk 4]
```diff
@@ -42,6 +50,10 @@ class CommentService {
     });
     return getLikes?.[0].likes || [];
```
Comment (Matt-Tan15) [inline src/services/comment.service.ts:51]: Is this checking if getLikes is empty?

**Review by raylu511 — APPROVED** [review]

**Review by raylu511 — APPROVED** [review]

**Review by ZohaibManzoor00 — APPROVED** [review]
Looks good

**Review by Matt-Tan15 — APPROVED** [review]

**Pull request merged**